bin/makeDocs.bash

```diff
@@ -3,118 +3,72 @@
 source colors.bash
 source lib.bash
 
-PWD="$(pwd)"
 
-# Get the path to the main repo directory.
-SCRIPT_PATH=$(readlink -f "$0")                     # Path to this script.
-SCRIPT_DIR=$(dirname "$SCRIPT_PATH")                # Path to directory containing this script.
-REPO_ROOT_DIR=$(builtin cd "$SCRIPT_DIR/.." && pwd) # REPO root directory.
-
-echo -e "${GREEN}Generating FarmData2 documentation.${NO_COLOR}."
+# Setting Env Variables
+PWD="$(pwd)" #Current Working Dir
+SCRIPT_PATH=$(readlink -f "$0") # Script Path
+SCRIPT_DIR=$(dirname "$SCRIPT_PATH") # Script Directory
+REPO_ROOT_DIR=$(builtin cd "$SCRIPT_DIR/.." && pwd) #Repository Root Directory
+
+TARGETS=()
+
+# Parsing all command line arguments
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --target) TARGETS+=("$2"); shift ;;
+        *) echo "Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
 
-echo "  Deleting old docs..."
-rm -rf docs/components 2 &> /dev/null
-mkdir docs/components
-rm -rf docs/library 2 &> /dev/null
-mkdir docs/library
-echo "  Deleted."
+INDEX_FILE="FarmData2.md" 
+INDEX_PATH="$REPO_ROOT_DIR/docs/$INDEX_FILE"
 
-echo "  Creating index file: $INDEX_FILE..."
-INDEX_FILE="FarmData2.md"
-INDEX_PATH="$REPO_ROOT_DIR/docs/$INDEX_FILE" # Absolute path.
+# Recreate the index file (Erases and rewrite into header upon regeneration)
 echo "# FarmData2 Documentation" > "$INDEX_PATH"
 echo "" >> "$INDEX_PATH"
-echo "  Created."
 
-echo "  Generating docs for all Components..."
-echo "    Adding Components section to $INDEX_FILE..."
+update_docs() {
+    local TYPE=$1 # components or library
+    local NAME=$2
+    local PATH="$REPO_ROOT_DIR/$TYPE/$NAME"
+    local DOC_PATH="docs/$TYPE/$NAME.md"
+
+    if [[ -d "$PATH" ]]; then
+        echo "    Generating docs for $NAME..."
+        # Use vue-docgen or jsdoc2md based on type
+        if [ "$TYPE" == "components" ]; then
+            npx vue-docgen "$PATH/$NAME.vue" -o "$REPO_ROOT_DIR/$DOC_PATH"
+        else
+            npx jsdoc2md "$PATH/$NAME.js" > "$REPO_ROOT_DIR/$DOC_PATH"
+        fi
+        echo "      Docs generated for $NAME."
+
+        # Adds link to the index file
+        echo "- [$NAME]($DOC_PATH)" >> "$INDEX_PATH"
```

Comment:
The links seem like they may be off... when visit the `FarmData2.md` page there are links for each component and library. However clicking on any of the links results in something like:

<img width="690" alt="image" src="https://github.com/FarmData2/FarmData2/assets/16962482/37a16c7c-424c-4ea2-9c97-0ee958e03192">

There seems to be an extra `docs` in there.

Reply:
Fixed Doc Path to be `local DOC_PATH="$TYPE/$NAME.md"`